XXX When werror=true is set, Meson passes the appropriate fatal-warnings flag to
the linker. We can't use it widely because of recurring:
warning: str*() is almost always misused, please use strl*()
Handle IABSD ORIGIN recording like OpenBSD.

Index: mesonbuild/linkers/linkers.py
--- mesonbuild/linkers/linkers.py.orig
+++ mesonbuild/linkers/linkers.py
@@ -734,7 +734,7 @@ class GnuLikeDynamicLinkerMixin(DynamicLinkerBase):
         return self._apply_prefix('--no-undefined')
 
     def fatal_warnings(self) -> T.List[str]:
-        return self._apply_prefix('--fatal-warnings')
+        return []
 
     def get_soname_args(self, prefix: str, shlib_name: str, suffix: str,
                         soversion: str, darwin_versions: T.Tuple[str, str]
@@ -1574,7 +1574,7 @@ class ClangClDynamicLinker(VisualStudioLikeLinkerMixin
         return ["/lldltocache:" + path]
 
     def fatal_warnings(self) -> T.List[str]:
-        return ['-WX']
+        return []
 
 
 class XilinkDynamicLinker(VisualStudioLikeLinkerMixin, DynamicLinker):
@@ -769,7 +769,7 @@
             all_paths.update(extra_paths)

         # TODO: should this actually be "for (dragonfly|open)bsd"?
-        if mesonlib.is_dragonflybsd() or mesonlib.is_openbsd():
+        if mesonlib.is_dragonflybsd() or mesonlib.is_openbsd() or mesonlib.is_iabsd():
             # This argument instructs the compiler to record the value of
             # ORIGIN in the .dynamic section of the elf. On Linux this is done
